src/common/expandingPanel/ExpandingPanel.tsx

```diff
@@ -41,9 +46,16 @@ function ExpandingPanel({
     container.current = ref;
   };
 
-  const { isOpen, buttonProps, contentProps } = useAccordion({
+  const { isOpen, buttonProps, contentProps, openAccordion } = useAccordion({
     initiallyOpen,
   });
+
+  useLayoutEffect(() => {
+    if (initiallyOpen && isOpen !== initiallyOpen) {
+      openAccordion();
+    }
+  }, [initiallyOpen, openAccordion, isOpen]);
```

Comment:
Almost, the panel cannot be closed from outside. If it is open, initiallyOpen === false, won't close it.

The internal state (close/open) of the ExpandingPanel should not be "exported" outside, or its state should be only outside. Making the component obsolete.

One way to do this would be to not render the Expanding panel at all, before the initiallyOpen is resolved. Problem is, there are other ExpandingPanels rendered and there would be flickering when the one to be deleted might be rendered later.

Reply:
Makes sense. So this new functionality could be described perhaps something like this: "for an already rendered ExpandingPanel, if it's currently closed, I want it to open (or if it's already open, nothing should happen)", right? From outside of the component, that sounds something like an action or an event, more than state. Is there a way to send "events" to components somehow? Something equivalent to calling a function perhaps?

Reply:
Yes, good description! It is a "force panel open" functionality. Just because the data, whether it should be open or not, is not available on first render.

Parent component has no access to child components functions. Child component does not exist, when parent is rendered. Profile UI uses functional components, so they have no instances to access. Usually parent passes callbacks to children and child can call those passed functions.  Child could call a passed function "shouldIBeOpen", but that is very unusual way to use React.

Javascript has events of course, but React does not use them for component-component communication. Except with Redux which has events. Kind of. It dispatches actions to state handlers. Parent and child can also share state and functions via React Context. But Redux and Context requires a shared data container, which is an overkill in this case.

Of course a window event listener could be added to the panel, and parent could emit an event. But that is not a common approach in React.

One solution would be to give React component a `key` which changes when the `initiallyOpen` changes. When component's `key`-prop changes, the old version is destroyed and a completely new instance is created with new state. But that solution has side-effects, because the panel is also closed when `initiallyOpen` changes. Unless a "openedOnce" is stored in the parent. :D But the Panel itself has that data, so it is left to that component to open itself, if `initiallyOpen` changes to `true`.

React has annoying quirks.

Reply:
Got it. Seems like there is no better way to achieve this.